user-service/src/main/java/hcmus/alumni/userservice/controller/UserServiceController.java

```diff
@@ -369,26 +380,99 @@ public ResponseEntity<String> adminCreateUser(@RequestBody UserModel req) {
-	@PutMapping("/{id}")
-	public ResponseEntity<String> adminUpdateUser(@PathVariable String id,
+	@PutMapping("/{id}/status")
+	public ResponseEntity<String> adminUpdateUserStatus(@PathVariable String id,
 			@RequestParam(value = "statusId", required = false) Integer statusId) {
-
 		Optional<UserModel> optionalUser = userRepository.findById(id);
 		if (!optionalUser.isPresent()) {
 			throw new AppException(20700, "Người dùng không tồn tại", HttpStatus.NOT_FOUND);
 		}
 
 		UserModel user = optionalUser.get();
 
-		if (statusId != null) {
-			user.setStatusId(statusId);
+		if (statusId == null) {
+			throw new AppException(20701, "Trạng thái tài khoản không được để trống", HttpStatus.BAD_REQUEST);
 		}
+
+		boolean statusExists = statusUserGroupRepository.existsById(statusId);
+		if (!statusExists) {
+			throw new AppException(20702, "Trạng thái tài khoản không tồn tại", HttpStatus.BAD_REQUEST);
+		}
+
+		user.setStatusId(statusId);
 		userRepository.save(user);
 
-		return ResponseEntity.status(HttpStatus.OK).body("User updated successfully");
+		return ResponseEntity.status(HttpStatus.OK).body("");
+	}
+
+	@PreAuthorize("hasAnyAuthority('User.Edit')")
+	@PutMapping("/{id}/role")
+	public ResponseEntity<String> adminUpdateUserRole(@PathVariable String id,
+			@RequestParam(value = "roleId", required = false) Integer roleId) {
+
+		Optional<UserModel> optionalUser = userRepository.findById(id);
+		if (!optionalUser.isPresent()) {
+			throw new AppException(20800, "Người dùng không tồn tại", HttpStatus.NOT_FOUND);
+		}
+
+		UserModel user = optionalUser.get();
+		if (roleId == null) {
+			throw new AppException(20801, "Vai trò không được để trống", HttpStatus.BAD_REQUEST);
+		}
+
+		RoleModel role = roleRepository.findById(roleId)
+				.orElseThrow(() -> new AppException(20802, "Vai trò không tồn tại", HttpStatus.BAD_REQUEST));
+		Set<RoleModel> roles = new HashSet<>();
+		roles.add(role);
+		user.setRoles(roles);
+		userRepository.save(user);
+
+		return ResponseEntity.status(HttpStatus.OK).body("");
+	}
+	
+	@GetMapping("/count/role")
+	public ResponseEntity<Long> getSearchResultCount(@RequestParam(value = "roleId", defaultValue = "0") Integer roleId) {
+		try {
+			if (roleId.equals(0)) {
+				return ResponseEntity.status(HttpStatus.OK).body(userRepository.countAllUsers());
+			}
+            return ResponseEntity.status(HttpStatus.OK).body(userRepository.countUsersByRoleId(roleId));
+        } catch (Exception e) {
+        	throw new AppException(20903, "Lỗi khi lấy số lượng người dùng. Vui lòng thử lại", HttpStatus.BAD_REQUEST);
+        }
+	}
+
+	@GetMapping("")
+	public ResponseEntity<HashMap<String, Object>> getSearchResult(
+			@RequestParam(value = "page", required = false, defaultValue = "0") int page,
+			@RequestParam(value = "pageSize", required = false, defaultValue = "10") int pageSize,
+			@RequestParam(value = "orderBy", required = false, defaultValue = "lastLogin") String orderBy,
```

Comment:
```suggestion
			@RequestParam(value = "orderBy", required = false, defaultValue = "fullName") String orderBy,
```
mặc định theo tên đi